### PR TITLE
Backend: incompatible redpanda version error

### PR DESCRIPTION
**Pull request by bojand**
This PR adds incompatible redpanda version error to common errors.

### CODE REVIEW

File: proto/redpanda/api/dataplane/v1alpha1/error.proto [hunk 1]
```diff
@@ -11,4 +11,5 @@ enum Reason {
   REASON_KAFKA_CONNECT_API_ERROR = 5;
   REASON_TYPE_MAPPING_ERROR = 6;
   REASON_SECRET_STORE_ERROR = 7;
+  REASON_FEATURE_NOT_SUPPORTED = 8;
```
Comment (weeco) [inline proto/redpanda/api/dataplane/v1alpha1/error.proto:14]: I think this should go into the common proto package. We already have one similar enum for features: https://buf.build/redpandadata/common/file/main:redpanda/api/common/v1alpha1/common.proto#L41

File: backend/pkg/api/connect/errors/common_errors.go [hunk 2]
```diff
@@ -27,3 +27,14 @@ func NewRedpandaAdminAPINotConfiguredError() *connect.Error {
...
+// NewRedpandaFeatureNotSupportedError is a standard error to return if an endpoint
+// requires the Redpanda feature that is not supported with current running version and instance.
+func NewRedpandaFeatureNotSupportedError(feature string) *connect.Error {
+	return NewConnectError(
+		connect.CodeUnimplemented,
+		errors.New("redpanda version does not support feature "+feature),
+		NewErrorInfo(v1alpha1.Reason_REASON_FEATURE_NOT_SUPPORTED.String()),
+		NewHelp(NewHelpLinkConsoleReferenceConfig()),
+	)
+}
```
Comment (weeco) [inline backend/pkg/api/connect/errors/common_errors.go:40]: Maybe we can use a more generic error here instead? We already have something similar for Kafka API endpoints if the target cluster version does not implement a certain Kafka API version.

File: backend/pkg/api/connect/errors/common_errors.go [hunk 3]
```diff
@@ -27,3 +27,14 @@ func NewRedpandaAdminAPINotConfiguredError() *connect.Error {
 		NewHelp(NewHelpLinkConsoleReferenceConfig()),
 	)
 }
+
+// NewRedpandaFeatureNotSupportedError is a standard error to return if an endpoint
+// requires the Redpanda feature that is not supported with current running version and instance.
+func NewRedpandaFeatureNotSupportedError(feature string) *connect.Error {
+	return NewConnectError(
+		connect.CodeUnimplemented,
+		errors.New("redpanda version does not support feature "+feature),
```
Comment (weeco) [inline backend/pkg/api/connect/errors/common_errors.go:36]: We could put the feature into the metadata instead. I assume it's some sort of an enum?

File: backend/pkg/api/connect/errors/common_errors.go (hunk 2 above)
Comment (bojand) [reply backend/pkg/api/connect/errors/common_errors.go:40]: I am not sure I understand exactly what you mean by a more generic error? Can you elaborate?

**Review by weeco — APPROVED** [review]

**Pull request merged**